### ISSUE TITLE
Add option to disable `db.statement` in DB instrumentation

### ISSUE THREAD

**Issue by fbogsany** [open-telemetry/opentelemetry-ruby#736]
See https://github.com/open-telemetry/opentelemetry-specification/pull/1659

We should use the same configuration option in all affected instrumentation.

Affected instrumentation:

- [x] Dalli
- [x] LMDB
- [x] Mongo
- [x] MySQL2
- [x] Postgres
- [x] Redis

**Comment by robertlaurin** [comment]
As mentioned in the description we should use the same config option.

Currently we're a bit inconsistent with the configuration naming here see examples:
https://github.com/open-telemetry/opentelemetry-ruby/blob/9dc42f12f7231bc48e2702d390c54917293d9b65/instrumentation/pg/lib/opentelemetry/instrumentation/pg/instrumentation.rb#L28-L29
https://github.com/open-telemetry/opentelemetry-ruby/blob/9dc42f12f7231bc48e2702d390c54917293d9b65/instrumentation/redis/lib/opentelemetry/instrumentation/redis/instrumentation.rb#L24
https://github.com/open-telemetry/opentelemetry-ruby/blob/9dc42f12f7231bc48e2702d390c54917293d9b65/instrumentation/mysql2/lib/opentelemetry/instrumentation/mysql2/instrumentation.rb#L23

Settling on a consistent name will introduce breaking changes but I think its valuable in the long term.

I have the opinion that the enable/disable prefix is a bit wordy and could be dropped (this is true of other config options, some I introduced myself).

```rb
option :db_statement,             default: true, validate: :boolean
option :db_statement_obfuscation, default: true, validate: :boolean
```

**Comment by SomalianIvan** [comment]
👀

**Comment by fbogsany** [comment]
> I have the opinion that the enable/disable prefix is a bit wordy and could be dropped

I'm not sure I agree 😄. I tend to think boolean config options benefit from being clearly a predicate, e.g. `include_db_statement: false` is clearer than `db_statement: false`. On the other hand, `obfuscate_db_statement: false` is clearer than either `enable_db_statement_obfuscation: false` or `db_statement_obfuscation: false`. So my vote would be for:
```ruby
option :include_db_statement,   default: true, validate: :boolean
option :obfuscate_db_statement, default: true, validate: :boolean
```
Alternatively, these could be unified:
```ruby
option :db_statement, default: :include, validate: ->(opt) { %I[include omit obfuscate].include?(opt) }
```

**Comment by robertlaurin** [comment]
> I'm not sure I agree 

<img src="https://user-images.githubusercontent.com/23463253/127527347-7175db0f-0509-4003-863a-d2bf441216be.png" width="120" height="100">

> I tend to think boolean config options benefit from being clearly a predicate

Looking at your examples I agree, I think my issue was with `enable_` / `disable_` prefixes.  Having the option names clearly be predicates is the better route.

>Alternatively, these could be unified:

```rb
option :db_statement, default: :include, validate: ->(opt) { %I[include omit obfuscate].include?(opt) }
```
I think I prefer the alternative overall when thinking about this from the perspective of an application that may have multiple instrumentation libraries that have this option.

```rb
{
  'OpenTelemetry::Instrumentation::Redis'  => { db_statement: :include },
  'OpenTelemetry::Instrumentation::Mysql2' => { db_statement: :obfuscate },
  'OpenTelemetry::Instrumentation::LMDB'   => { db_statement: :omit },
}
```
It just seems clearer to understand.  In the example where they are split which one takes priority?  If I do not include the db.statement but enable obfuscation what should happen?  With the alternative this is no longer a concern.
